core/src/test/java/ai/timefold/solver/core/impl/heuristic/selector/move/generic/list/ListChangeMoveSelectorTest.java

```diff
@@ -445,10 +445,10 @@ void randomWithEntityValueRangeAndFiltering() {
         // and that causes the B destination to become an invalid destination
         assertCodesOfNeverEndingMoveSelector(moveSelector,
                 "1 {A[1]->A[1]}",
-                "3 {B[0]->A[0]}",
-                "3 {B[0]->A[0]}",
-                "3 {B[0]->A[2]}",
-                "3 {B[0]->A[1]}");
+                "3 {B[0]->B[0]}",
+                "3 {B[0]->B[0]}",
+                "3 {B[0]->B[0]}",
```

Comment:
That is a lot of pointless moves generated.
I wonder if we should change the test to actually check that something productive is created.

Reply:
The results are this way because the arrangement of the entity range is too restrictive. A small change in the value range of `A` completely alters the results.